src/lib.rs

```diff
@@ -262,6 +262,174 @@ impl<T, const N: usize> Vector<T, {N}> {
+                Vector::<T, 3>::from([
+                    self.$a(),
+                    self.$b(),
+                    self.$c(),
+                ])
+            }
+        }
+
+        // Do not need to pass the alphabet because the fourth level does not need to choose
+        // any more letters.
+        swizzle!{ $a, $b, $c, $x }
+        swizzle!{ $a, $b, $c, $y }
+        swizzle!{ $a, $b, $c, $z }
+        swizzle!{ $a, $b, $c, $w }
+    };
+    // Final level which halts the recursion. Generates all 4-element swizzle functions.
+    // No $x, $y, $z, $w parameters because this function does not need to know the alphabet,
+    // because it already has all the names assigned.
+    ($a:ident, $b:ident, $c:ident, $d:ident) => {
+        paste::item! {
+            fn [< $a $b $c $d >](&self) -> Vector<T, 4> {
+                Vector::<T, 4>::from([
+                    self.$a(),
+                    self.$b(),
+                    self.$c(),
+                    self.$d(),
+                ])
+            }
+        }
+    };
+}
+
+/// Provides [swizzling](https://en.wikipedia.org/wiki/Swizzling_(computer_graphics))
+/// support for up to four elements. Swizzling is a technique for easily
+/// rearranging and accessing elements of a vector, used commonly in graphics shader
+/// programming. Swizzling is available on vectors whose element type is Copy.
+///
+/// Single-element accessors return the element itself. Multi-element accessors return
+/// vectors of the appropriate size.
+///
+/// # Element names
+/// Only the first four elements of a vector may be swizzled. If you have vectors larger
+/// than length four and want to manipulate their elements, you must do so manually.
+///
+/// Because swizzling is often used in compute graphics contexts when dealing with colors,
+/// both 'xyzw' and 'rgba' element names are available. They cannot however be mixed.
+///
+/// | Element Index | xyzw Name | rgba Name |
+/// |---------------|-----------|-----------|
+/// | 0             | x         | r         |
+/// | 1             | y         | g         |
+/// | 2             | z         | b         |
+/// | 3             | w         | a         |
+///
+/// # Restrictions
+/// It is a compilation error to attempt to access an element beyond the bounds of a vector.
+/// For example, `vector![1, 2].z()` would fail because `z()` is only available on vectors of
+/// length 3 or greater.
+///
+/// # Examples
+///
+/// To get the first two elements of a 4-vector.
+/// ```ignore
+/// let v = vector![1, 2, 3, 4].xy();
+/// ```
+///
+/// To get the first and last element of a 4-vector.
+/// ```ignore
+/// let v = vector![1, 2, 3, 4].xw();
+/// ```
+///
+/// To reverse the order of a 3-vector.
+/// ```ignore
+/// let v = vector![1, 2, 3].zyx();
+/// ```
+///
+/// To select the first and third elements into the second and fourth elements, respectively.
+/// ```ignore
+/// let v = vector![1, 2, 3, 4].xxzz();
+/// ```
+pub trait Swizzle<T> {
```

Comment:
I believe that `Swizzle` must require `Sized` in order to make it object-unsafe. As of right now you can create Swizzle trait objects and I believe that could cause undefined behaviour.

Reply:
Interesting, what would the UB be? Understanding trait objects and object safety is a weak part in my Rust knowledge.

Adding `: Sized` bound.

Reply:
I don't think it's UB. Fortunately the rust designers figured this one out correctly.

My issue is that without the Sized bound you would be able to write the following: 
```rust
fn take_swizzle<T>(swiz: &dyn Swizzle<T>) -> T
where
    T: Copy
{
    swiz.z()
}
```
attempts to pass in a vec to this function with N < 3 causes an outs of bounds error as one would expect:
```index out of bounds: the len is 1 but the index is 2```
This error is not terribly helpful, so I think for now it would be wise to disallow this for now. 
Making this sized removes the ability to turn this into a trait object.